src/QsCompiler/QirGeneration/DebugInfoManager.cs

```diff
@@ -0,0 +1,585 @@
+        {
+            public uint Line { get; set; }
+
+            public uint Column { get; set; }
+
+            private DebugPosition(uint line, uint col)
+            {
+                this.Line = line;
+                this.Column = col;
+            }
+
+            public static DebugPosition FromZeroBased(uint line, uint col)
+            {
+                return new DebugPosition(line + 1, col + 1);
+            }
+
+            public static DebugPosition FromZeroBased(int line, int col)
+            {
+                return FromZeroBased((uint)line, (uint)col);
+            }
+
+            public static DebugPosition FromZeroBasedLine(uint line)
+            {
+                return new DebugPosition(line + 1, 1);
+            }
+
+            public static DebugPosition FromZeroBasedLine(int line)
+            {
+                return FromZeroBasedLine((uint)line);
+            }
+        }
+
+        /// <summary>
+        /// Converts from a 0-based <see cref="Position"/> to a 1-based <see cref="DebugPosition"/>.
+        /// </summary>
+        /// <param name="position">The 0-based <see cref="Position"/></param>
+        /// <return>The 1-based <see cref="DebugPosition"/></return>
+        private static DebugPosition ConvertToDebugPosition(Position position)
+        {
+            return DebugPosition.FromZeroBased(position.Line, position.Column);
+        }
+
+        public static string DebugTypeNotSupportedMessage => "This debug type is not yet supported";
+
+        /// <summary>
+        /// Contains the location information for the statement nodes we are currently parsing
+        /// </summary>
+        internal Stack<QsLocation?> StatementLocationStack { get; }
+
+        /// <summary>
+        /// Contains the location information for the Expression we are inside
+        /// </summary>
+        internal Stack<DataTypes.Range?> ExpressionRangeStack { get; }
+
+        /// <summary>
+        /// Contains the location information for the namespace element we are inside
+        /// </summary>
+        internal QsLocation? CurrentNamespaceElementLocation { get; set; }
+
+        /// <summary>
+        /// The GenerationContext that owns this DebugInfoManager
+        /// </summary>
+        private readonly GenerationContext sharedState;
+
+        /// <summary>
+        /// Contains the DIBuilders included in the module related to this DebugInfoManager.
+        /// The key is the source file path of the DebugInfoBuilder's compile unit
+        /// </summary>
+        private readonly Dictionary<string, DebugInfoBuilder> dIBuilders;
+
+        /// <summary>
+        /// Constructs a DebugInfoManager with access to the <see cref="GenerationContext"/> as the shared state.
+        /// </summary>
+        internal DebugInfoManager(GenerationContext generationContext, bool enableDebugSymbols)
+        {
+            this.sharedState = generationContext;
+            Config.EnableDebugSymbols = enableDebugSymbols;
```

Comment:
It is odd that an argument to a class instance construction is used to set a property in the Config that is no limited to this instance.